src/Agent/NewRelic/Agent/Core/JsonConverters/LogEventWireModelCollectionJsonConverter.cs

```diff
@@ -109,7 +109,25 @@ private static void WriteJsonImpl(JsonWriter jsonWriter, LogEventWireModelCollec
                     foreach (var item in logEvent.ContextData)
                     {
                         jsonWriter.WritePropertyName(Context + "." + item.Key);
-                        jsonWriter.WriteValue(item.Value);
+                        string contextValueJson;
+                        try
+                        {
+                            contextValueJson = JsonConvert.SerializeObject(item.Value);
+                        }
+                        catch
+                        {
+                            // If JsonConvert can't serialize it, maybe it has a ToString()
+                            try
+                            {
+                                contextValueJson = item.Value.ToString();
+                            }
+                            catch
+                            {
+                                // If that didn't work, just use the type name
+                                contextValueJson = item.Value.GetType().ToString();
```

Comment:
I don't really like just ToStringing something and dropping the value in even if its just as a fallback.  In other areas of the agent we use switches with specific actions for different types and for complex types that action is usually "drop".

Reply:
Discussed as a team and we agreed to keep the current behavior.